src/repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RepairnatorPostBuild.java

```diff
@@ -74,18 +83,44 @@
     private boolean useAstorJMut;
     private boolean useNPEFixSafe;
     private boolean useNopolTestExclusionStrategy;
+    private boolean useSonarQubeRepair;
     private final Config config = new Config();
 
+    private boolean sonarRulesGiven;
+    /* Rules */
+    private boolean rule1656;
```

Comment:
avoid boolean, prefer enum.

What about "List<SonarQubeRule>"?

Reply:
This is a convention of how we should define each field for fetching into the `.jelly` (the UI file). I can make a list contains all the fields but the setter and getter need to remains otherwise your checkboxes will be unmarked in the config of Jenkins Job, despite being marked and saved before :) .

Reply:
I just found out that this will not work with a list (I tried a HashMap). The field name need to be `rule1000` for instance then you need a setter in DescriptorImpl (the UI part) called `setRule1000` and `getRule1000` outside the descriptor . The `.jelly` infer the getter and setter name from the field name to call the correct method (example in jelly below, if we apply after specifying port it will look up getPort and setPort). Well , this means that if we have a single list as a single field then, we can't decide which to get. I tried to look for some event based stuffs to distinct the cases but the documentation of how to make a plugin is so bad. 

```
<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
  <f:entry title="${%Port}" field="port">
    <f:textbox default="80" />
  </f:entry>
</j:jelly>
```